### PR TITLE
Rename submodules under `cupy.lib`

### PR DESCRIPTION
**Pull request by takagi**
Rel #3420.

This PR renames `cupy.lib` submodule to `cupy._lib` to indicate it is CuPy's private submodule and is not intended to be used by users.

### CODE REVIEW

**Comment by asi1024** [comment]
@takagi Could you resolve conflicts?

**Comment by takagi** [comment]
I've resolved the conflicts and turned it ready for review!

**Comment by asi1024** [comment]
Could you check Travis failure?

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by takagi** [comment]
pfnCI, test this please.

**Comment by asi1024** [comment]
@takagi Could you check the pretest failure?

**Comment by takagi** [comment]
Maybe we should keep `cupy.lib`?

```Python
>>> import numpy
>>> numpy.lib
<module 'numpy.lib' from '/home/ext-mtakagi/miniconda3/lib/python3.9/site-packages/numpy/lib/__init__.py'>
```

**Comment by asi1024** [comment]
Oh, `numpy.lib.stride_tricks.as_strided` is documented, so we should keep `cupy.lib`.
https://numpy.org/doc/stable/reference/generated/numpy.lib.stride_tricks.as_strided.html

However, `numpy.lib.poly1d` does not exist. How about polynomial routines to `_polynomial` module?

**Comment by takagi** [comment]
Sounds good. I made changes as so.

**Comment by takagi** [comment]
pfnCI, test this please.

File: cupy/lib/__init__.py [hunk 1]
```diff
@@ -1,8 +0,0 @@
-from cupy.lib import stride_tricks  # NOQA
```
Comment (asi1024) [inline cupy/lib/__init__.py:1]: It is a documented interface, so it should not be removed.

Comment (takagi) [reply cupy/lib/__init__.py:1]: Fixed!

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
LGTM.

**Pull request merged**